### DISCUSSION TITLE
Using regex for filter database

### DISCUSSION THREAD

**Discussion by iBot42** [Budibase/budibase#13149] [Help]
Hello,
I have a problem to filter for input data created by a csv import. I did read the docs on filtering and bindings , but could not achive what I need.
The csv data is a big list with userinformation like profilenames , userhandles, mailadresses. Some data does not deal with the right namingconvention ( lowercase / uppercase / sequence / etc.) . I want to filterout these and display them as table with a export button.

The filtering on the budibase_db does not support regex . And there is also no support of placeholders or wildcards ? 
Next try was on the desing side and use a filter on my columns with a binding and a javascript function like shown below, but I get no output at my table.

```
let wireusers = $("wUsers_emptyHandle - Table block.Selected rows") ;
let out = wireusers.search(/_regex_/i) ;
$("wUsers_emptyHandle - Table block.Selected rows") = out ;
return $("wUsers_emptyHandle - Table block.Selected rows") ;

```
Any help or ideas or maybe an example ? 
Thanks an Regards

**Comment by melohagan** [comment]
Hey @iBot42 

> $("wUsers_emptyHandle - Table block.Selected rows") = out ;

The bindings are readonly. You cannot assign values to them. Furthermore, search applied to text (strings), whereas the `Selected Rows` binding is an array. You may need to use the `join` function first.

**Comment by iBot42** [comment]
@melohagan 
thanks for pointing me on that. But I did not get it managed to work. I did not even got output in the debugger console of the webbrowser by doing something like this.
```
let text = $("TEST Table Data.Selected rows") ;
console.log({
  text
})
```
  
I Would suggest for the roadmap it would be nice to just have an regex operator for the filtering of data inside the data or design view. No need for scripting.

Regards
